src/sentry/api/fields/actor.py

```diff
@@ -18,7 +21,7 @@ def to_internal_value(self, data):
                 "Could not parse actor. Format should be `type:id` where type is `team` or `user`."
             )
         try:
-            obj = actor.resolve()
+            obj: APIUser | Team = actor.resolve()
```

Comment:
Wherever I thought it was helpful to really deeply annotate what's going on, I added more annotations.